testbed/tests/window/test_window.py

```diff
@@ -507,6 +662,227 @@ async def test_move_and_resize(second_window, second_window_probe):
+        # Add delay to ensure windows are visible after animation.
+        await second_window_probe.wait_for_window(
+            f"Secondary window is in {initial_state}",
+            minimize=True if initial_state == WindowState.MINIMIZED else False,
+            full_screen=True if initial_state == WindowState.FULLSCREEN else False,
+        )
+        assert second_window_probe.instantaneous_state == initial_state
+
+        # Set to final state
+        second_window.state = final_state
+        # Add delay to ensure windows are visible after animation.
+        await second_window_probe.wait_for_window(
+            f"Secondary window is in {final_state}",
+            state_switch_not_from_normal=(
+                True if initial_state != WindowState.NORMAL else False
+            ),
+        )
+        assert second_window_probe.instantaneous_state == final_state
+
+    @pytest.mark.parametrize(
+        "states",
+        [
+            # Switch from NORMAL:
+            (WindowState.NORMAL, WindowState.MINIMIZED),
+            (WindowState.NORMAL, WindowState.MAXIMIZED),
+            (WindowState.NORMAL, WindowState.FULLSCREEN),
+            (WindowState.NORMAL, WindowState.PRESENTATION),
+            (WindowState.NORMAL, WindowState.NORMAL),
+            # Switch from MINIMIZED:
+            (WindowState.MINIMIZED, WindowState.NORMAL),
+            (WindowState.MINIMIZED, WindowState.MAXIMIZED),
+            (WindowState.MINIMIZED, WindowState.FULLSCREEN),
+            (WindowState.MINIMIZED, WindowState.PRESENTATION),
+            (WindowState.MINIMIZED, WindowState.MINIMIZED),
+            # Switch from MAXIMIZED:
+            (WindowState.MAXIMIZED, WindowState.NORMAL),
+            (WindowState.MAXIMIZED, WindowState.MINIMIZED),
+            (WindowState.MAXIMIZED, WindowState.FULLSCREEN),
+            (WindowState.MAXIMIZED, WindowState.PRESENTATION),
+            (WindowState.MAXIMIZED, WindowState.MAXIMIZED),
+            # Switch from FULLSCREEN:
+            (WindowState.FULLSCREEN, WindowState.NORMAL),
+            (WindowState.FULLSCREEN, WindowState.MINIMIZED),
+            (WindowState.FULLSCREEN, WindowState.MAXIMIZED),
+            (WindowState.FULLSCREEN, WindowState.PRESENTATION),
+            (WindowState.FULLSCREEN, WindowState.FULLSCREEN),
+            # Switch from PRESENTATION:
+            (WindowState.PRESENTATION, WindowState.NORMAL),
+            (WindowState.PRESENTATION, WindowState.MINIMIZED),
+            (WindowState.PRESENTATION, WindowState.MAXIMIZED),
+            (WindowState.PRESENTATION, WindowState.FULLSCREEN),
+            (WindowState.PRESENTATION, WindowState.PRESENTATION),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "second_window_class, second_window_kwargs",
+        [
+            (
+                toga.MainWindow,
+                dict(title="Secondary Window", position=(200, 150)),
+            )
+        ],
+    )
+    async def test_window_state_rapid_assignment(
```

Comment:
This new test ensures complete coverage of this PR on macOS and gtk. So, I have removed all `#pragma: no cover` from this PR.

Reply:
ok... but it also seems like a direct reproduction of the previous test, but without the (implied) delay between state changes. 

Prior to this PR, the testbed takes ~3:20 to complete on macOS ARM64. As of the most recent CI run on this PR, it took almost 7 minutes to run. Window state changes aren't fast - and there's a *lot* of tests needed here. 

For a backend that doesn't have any delay issues (like Windows), this test is `identical` to the window state change test. And for some macOS/GTK states, it will also be identical.

In an ideal world - yes, we'd run all these tests, reflecting every possible permutation "done fast". However, these tests are *slow*, so we need to ensure we're only testing what we need to test.

There's even an argument to be made that the two tests are the same test, but with some transitions having a boolean "...and wait" condition between the two state changes.